test/cppsuite/test_harness/workload/thread_context.cpp

```diff
@@ -109,6 +109,19 @@ transaction_context::commit(const std::string &config)
 {
     WT_DECL_RET;
     testutil_assert(_in_txn && !_needs_rollback);
+
+    /*
+     * We may have set the commit ts to a value that is now older than the current stable timestamp,
+     * we need to rollback as it is an invalid transaction. TODO - Should we have this check here or
```

Comment:
Happy to get some ideas here regarding the `TODO`.

Reply:
I'm in favour of a single common check here. I can't see this behaviour needing to differ across operations.

Reply:
I'm with @ajmorton here

Reply:
Thanks, updated in efd54370a5f8fae9703829ea1ffb39382a5ba6c8